src/components/post-comments.jsx

```diff
@@ -57,8 +57,13 @@ export default (props) => {
       props.toggleCommenting(props.post.id);
     }
 
-    const updatedCommentText = `@${username} ` + (props.post.newCommentText || '');
-    props.updateCommentingText(props.post.id, updatedCommentText);
+    const text = (props.post.newCommentText || '');
+    const check = `@${username}$`;
+
+    if (!text.match(check)) {
```

Comment:
Checked in chrome, all works. Looks like it's converting param to regex under the hood or something